### PR TITLE
Fix EXIF orientation for iPhone photo uploads

### PR DESCRIPTION
**Pull request by Angak0k**
## Summary
- iPhone photos taken in portrait appear rotated to landscape because Go's `image.Decode()` ignores EXIF orientation metadata
- Added EXIF orientation parsing (`readEXIFOrientation`) and pixel-level transform (`applyOrientation`) in `pkg/images/orientation.go`
- Integrated into `DecodeImage()` so all 4 image processing pipelines (pack, profile, inventory, banner) are fixed in one place
- Rotation applied before resize for correct dimension handling
- Comprehensive unit tests covering all 8 EXIF orientations, both endianness, and edge cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/images/orientation.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			return 1
+		}
+
+		// APP1 marker (0xE1) — EXIF data
+		if marker == 0xE1 {
+			return parseEXIFOrientation(data[offset+4 : offset+2+segLen])
+		}
+
+		// Skip to next marker
+		offset += 2 + segLen
```
[the remark this answers is not in the document]
Comment (Angak0k) [reply pkg/images/orientation.go]: Fixed — now checks for `"Exif\x00\x00"` header before treating an APP1 as EXIF, so XMP and other non-EXIF APP1 segments are skipped. Also added a test with an XMP APP1 preceding the EXIF one.

File: pkg/images/orientation_test.go [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+			}{
+				{0, 0, white}, {1, 0, black}, {0, 3, red}, {1, 3, yellow},
+			},
+		},
+	}
+
+	_ = green
+	_ = blue
+	_ = cyan
+	_ = magenta
```
[the remark this answers is not in the document]
Comment (Angak0k) [reply pkg/images/orientation_test.go]: Removed the unused color variables and the blank identifier assignments.

File: pkg/images/orientation_test.go [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+				}
+				if got != check.c {
+					t.Errorf("pixel(%d,%d) = %v, want %v", check.x, check.y, got, check.c)
+				}
+			}
+		})
+	}
+}
+
+// TestProcessImage_EXIFOrientation6 verifies the full pipeline correctly rotates
```
[the remark this answers is not in the document]
Comment (Angak0k) [reply pkg/images/orientation_test.go:381]: Fixed — clarified the comments to unambiguously state stored pixel dimensions and expected post-correction dimensions.

File: pkg/images/orientation.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+			return 1
+		}
+		segLen := int(binary.BigEndian.Uint16(data[offset+2 : offset+4]))
+		if segLen < 2 {
+			return 1
+		}
+
+		// APP1 marker (0xE1) — EXIF data
+		if marker == 0xE1 {
+			return parseEXIFOrientation(data[offset+4 : offset+2+segLen])
```
[the remark this answers is not in the document]
Comment (Angak0k) [reply pkg/images/orientation.go]: Fixed — added `segEnd` bounds check before slicing to prevent panic on truncated/malformed JPEGs.
